lib/core/src/base64.cpp

```diff
@@ -1,138 +1,120 @@
-                   unsigned char *out, unsigned long *outlen ) {
-    unsigned long t, x, y, z;
-    unsigned char c;
-    int           g;
+    int base64_decode(const unsigned char* in, unsigned long inlen, unsigned char* out, unsigned long* outlen)
+    {
+        unsigned long t, x, y, z;
+        unsigned char c;
+        int g;
 
-    g = 3;
-    for ( x = y = z = t = 0; x < inlen; x++ ) {
-        c = mymap[in[x] & 0xFF];
-        if ( c == 255 ) {
-            continue;
-        }
-        /* the final = symbols are read and used to trim the remaining bytes */
-        if ( c == 254 ) {
-            c = 0;
-            /* prevent g < 0 which would potentially allow an overflow later */
-            if ( --g < 0 ) {
+        g = 3;
+        for (x = y = z = t = 0; x < inlen; x++) {
+            c = mymap[in[x] & 0xFF];
+            if (c == 255) {
+                continue;
+            }
+            /* the final = symbols are read and used to trim the remaining bytes */
+            if (c == 254) {
+                c = 0;
+                /* prevent g < 0 which would potentially allow an overflow later */
+                if (--g < 0) {
+                    return BASE64_INVALID_PACKET;
+                }
+            }
+            else if (g != 3) {
+                /* we only allow = to be at the end */
                 return BASE64_INVALID_PACKET;
             }
-        }
-        else if ( g != 3 ) {
-            /* we only allow = to be at the end */
-            return BASE64_INVALID_PACKET;
-        }
 
-        t = ( t << 6 ) | c;
+            t = (t << 6) | c;
 
-        if ( ++y == 4 ) {
-            if ( z + g > *outlen ) {
-                return BASE64_BUFFER_OVERFLOW;
-            }
-            out[z++] = ( unsigned char )( ( t >> 16 ) & 255 );
-            if ( g > 1 ) {
-                out[z++] = ( unsigned char )( ( t >> 8 ) & 255 );
-            }
-            if ( g > 2 ) {
-                out[z++] = ( unsigned char )( t & 255 );
+            if (++y == 4) {
+                if (z + g > *outlen) {
+                    return BASE64_BUFFER_OVERFLOW;
+                }
+                out[z++] = (unsigned char) ((t >> 16) & 255);
+                if (g > 1) {
+                    out[z++] = (unsigned char) ((t >> 8) & 255);
+                }
+                if (g > 2) {
+                    out[z++] = (unsigned char) (t & 255);
+                }
+                y = t = 0;
             }
-            y = t = 0;
         }
+        if (y != 0) {
+            return BASE64_INVALID_PACKET;
+        }
+        *outlen = z;
+        return 0;
     }
-    if ( y != 0 ) {
-        return BASE64_INVALID_PACKET;
-    }
-    *outlen = z;
-    return OK;
+} // namespace irods
+
+// For backwards compatibility we are keeping but deprecating
+// the methods that are not in the irods namespace.
+int base64_encode(const unsigned char* in, unsigned long inlen, unsigned char* out, unsigned long* outlen)
```

Comment:
Should we mark the declarations in the header file with `__attribute__(deprecated)` as well?

Reply:
I did this.  Please take a look to make sure it is done correctly.  I have never used that.